server/sandbox_run_linux.go

```diff
@@ -427,7 +427,15 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 		return nil, err
 	}
 
-	kubeAnnotations := sbox.Config().Annotations
+	kubeAnnotations, err := s.Runtime().RuntimeDefaultAnnotations(runtimeHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	// override default annotations with pod spec specified ones
+	for k, v := range sbox.Config().Annotations {
+		kubeAnnotations[k] = v
```

Comment:
We probably should warn here.

Reply:
hmm I'm hesitant. On the one hand mentioning could be useful, but logging it may expose data about the workloads to the journal, and also could get noisy

Reply:
are you thinking something like warning `"Pod %s has default annotation %s that will be replaced by %s", sbox.ID(), k, v`

Reply:
Yeah, but we can also omit it. I was just thinking to make the behavior clear to end users. Could be also a debug log, though.

Reply:
Yeah, could be a simple message "Overriding default pod annotation %q with %q" just to leave a breadcrumb for users to follow if they are surprised to what is going on.

Reply:
Yeah, I think it make sense to display this as a debug message.

Reply:
for some reason I'm anxious about leaving the value of the overriden env in case there's a secret or sensitive information in the value, so I just printed overriding key, does that work for y'all @saschagrunert @sohankunkerkar @kwilczynski

Reply:
@haircommander sounds good!